### DISCUSSION TITLE
Does pytorch-lightning work with Pytorch 2.0?

### DISCUSSION THREAD

**Discussion by eraoul** [Lightning-AI/pytorch-lightning#15898] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I know Pytorch 2 was just announced and it's not an official release yet, but can we use it already with lightning?

**Comment by johnnynunez** [comment]
any news?

**Reply by akihironitta** [reply]
The PyTorch 2.0 feature `torch.compile(lightning_model)` should be available in master, which was tracked in #15894.

**Comment by SonodaKazuto** [comment]
You can check [this link](https://github.com/Lightning-AI/lightning/issues/15894) but actually I face some problems.

**Reply by johnnynunez** [reply]
My problem is AssertionError: expected FunctionType found method <bound method MNISTLitModule.forward of MNISTLitModule(
https://github.com/johnnynunez/lightning-hydra-template

**Reply by SonodaKazuto** [reply]
Yah, I also encountered this problem but I don't know how to fix it.

**Reply by johnnynunez** [reply]
@williamFalcon

**Reply by akihironitta** [reply]
hey @SonodaKazuto @johnnynunez! if you see any issue around the PyTorch 2.0 support, mind creating a GitHub issue? (#15894 enabled the PyTorch 2.0 feature in master.)

**Reply by SonodaKazuto** [reply]
Yah, I think that will be good.

**Reply by AugustoPeres** [reply]
Did anyone solve this? I am having the same issue but did not find any answers in #15894

**Reply by akihironitta** [reply]
@AugustoPeres What issue are you facing? Mind asking in our forum [here](https://lightning.ai/forums/)?

**Reply by AugustoPeres** [reply]
@akihironitta thanks for your reply.:

```python
model = MyModule() # this is a ligthning Module
model = torch.compile(model)

trainer = Trainer(...)
trainer.fit(model, ...) 
```

I then get the error:
```
AssertionError: expected FunctionType found method
```


Versions:

```
>>> import torch
>>> import pytorch_lightning
>>> torch.__version__
'2.0.0.dev20230116+cu117'
>>> pytorch_lightning.__version__
'1.8.6'
```

Am I not allowed to compile `LightningModules`? From #15894 I was under the impression that I could.

**Reply by akihironitta** [reply]
@AugustoPeres Mind installing PL from master and giving it a try again? I think the feature may not have been released yet.

**Reply by AugustoPeres** [reply]
@akihironitta Tried the following commands:

```
pip install https://github.com/Lightning-AI/lightning/archive/refs/heads/master.zip -U
```

```
pip install https://github.com/Lightning-AI/lightning/archive/refs/heads/release/stable.zip -U
```

with the same results. I think that the feature should already be present, no? [1.8.4 release note](https://github.com/Lightning-AI/lightning/releases/tag/1.8.4) and [pull request implementing the feature](https://github.com/Lightning-AI/lightning/pull/15922)

**Reply by akihironitta** [reply]
@AugustoPeres Thanks for trying it out :) By the way, what's the complete traceback that you've pasted above?
> ```
> AssertionError: expected FunctionType found method
> ```

It would be fantastic if you could create a new GitHub issue with your script and env detail for repro!

**Reply by johnnynunez** [reply]
In my case is using hydra and lightning.

**Reply by akihironitta** [reply]
Hi @johnnynunez please create a GitHub issue with your script and env detail! Thank you

**Reply by johnnynunez** [reply]
> Hi @johnnynunez please create a GitHub issue with your script and env detail! Thank you

opened https://github.com/Lightning-AI/lightning/issues/16395

**Reply by AugustoPeres** [reply]
> > Hi @johnnynunez please create a GitHub issue with your script and env detail! Thank you
> 
> opened #16395

@akihironitta  and @johnnynunez this is exactly what was happening with me so I will refrain from opening another issue and will wait for an answer there.

**Comment by akihironitta** [comment]
hey @eraoul! PyTorch Lightning already supports PyTorch 2.0 as PyTorch 2.0 is fully backwards-compatible. The major PyTorch 2.0 feature `torch.compile(model)` should work in master, too, according to the tracking issue https://github.com/Lightning-AI/lightning/issues/15894#issuecomment-1336604958. If you find any issues, please contribute by creating GitHub issues! ⚡
